src/Forms/GridField/GridFieldExportButton.php

```diff
@@ -167,6 +167,17 @@ public function generateExportFileData($gridField)
         $csvWriter->setEnclosure($this->getCsvEnclosure());
         $csvWriter->setNewline("\r\n"); //use windows line endings for compatibility with some csv libraries
         $csvWriter->setOutputBOM(Writer::BOM_UTF8);
+        $csvWriter->addFormatter(function (array $row) {
+            if (!Config::inst()->get(get_class($this), 'xls_export_disabled')) {
```

Comment:
You could put this config condition on the outside of the `addFormatter` call to cut down a large number of config api calls.

Reply:
Done